src/template-compiler/index.js

```diff
@@ -0,0 +1,68 @@
+const compiler = require('vue-template-compiler')
+const transpile = require('vue-template-es2015-compiler')
+const { js_beautify: beautify } = require('js-beautify')
+
+const transformRequire = require('./modules/transform-require')
+
+module.exports = function compileTemplate (template, filename, config) {
+  const options = Object.assign({
+    preserveWhitespace: true
+  }, config.options, {
+    modules: [transformRequire(config.transformToRequire)].concat(config.plugins || [])
+  })
+  const compile = (config.isServer && config.optimizeSSR !== false && compiler.ssrCompile) ? compiler.ssrCompile : compiler.compile
+  const compiled = compile(template.code, options)
+  const output = {
+    errors: compiled.errors,
+    tips: compiled.tips
+  }
+
+  if (output.errors && output.errors.length) {
+    output.code = config.esModule !== false
+      ? `export function render () {}\nexport var staticRenderFns = []`
+      : 'module.exports={render:function(){},staticRenderFns:[]}'
+  } else {
+    output.code = transpile(
+      'var render = ' + toFunction(compiled.render) + '\n' +
+      'var staticRenderFns = [' + compiled.staticRenderFns.map(toFunction).join(',') + ']',
+      config.buble
+    ) + '\n'
+
+    // mark with stripped (this enables Vue to use correct runtime proxy detection)
+    if (
+      !config.isProduction && (
+      !config.buble ||
+      !config.buble.transforms ||
+      config.buble.transforms.stripWith !== false
+    )) {
+      output.code += `render._withStripped = true\n`
+    }
+
+    const __exports__ = `{ render: render, staticRenderFns: staticRenderFns }`
+    output.code += config.esModule !== false
+      ? `export default ${__exports__}`
+      : `module.exports = ${__exports__}`
+
+    if (!config.isProduction && config.isHot) {
+      output.code +=
+        '\nif (module.hot) {\n' +
+        '  module.hot.accept()\n' +
+        '  if (module.hot.data) {\n' +
+        `     require('vue-hot-reload-api').rerender('${options.scopeId}', module.exports)\n` +
+        '  }\n' +
+        '}'
+    }
+  }
+
+  return Promise.resolve(output)
```

Comment:
According to the docs, Jest needs synchronous compilation

Reply:
Does callback work?

Reply:
I don't think so, I'll need to check. The docs say the transformers need to be synchronous

Reply:
Hey! Can you point me to the transformer docs? I'll experiment with jest and async transformer.

Reply:
Or it could be sync if there is sync API for [`postcss.process`](http://api.postcss.org/Processor.html#process)

Reply:
>A transformer is a module that provides a synchronous function for transforming source files

https://facebook.github.io/jest/docs/en/configuration.html#transform-object-string-string

Reply:
We could provide a synchronous version by default and an async wrapper for template-compiler. 

Looks like we can do the same with style-compiler - https://github.com/postcss/postcss/issues/595.

You can call .css on the result of process instead of .then. This makes it act sycnhronously